src/main.rs

```diff
@@ -17,110 +17,76 @@
 /// If only the path is given, this will generate a summary report of any claims in that file
 /// If a manifest definition json file is specified, the claim will be added to any existing claims
 ///
-use std::fs::{create_dir_all, remove_dir_all};
-use std::path::{Path, PathBuf};
+use std::fs::{create_dir_all, remove_dir_all, File};
+use std::io::Write;
+use std::path::PathBuf;
 
 use anyhow::{anyhow, bail, Context, Result};
-use c2pa::{Error, ManifestStore, ManifestStoreReport};
-use structopt::{clap::AppSettings, StructOpt};
+use c2pa::{Error, Ingredient, Manifest, ManifestStore, ManifestStoreReport};
+use clap::{AppSettings, Parser};
 
 mod info;
 use info::info;
-pub mod manifest_config;
-use manifest_config::ManifestConfig;
-mod fs_report;
 mod signer;
-use signer::get_c2pa_signer;
+use signer::SignConfig;
 
-// define the command line options
-#[derive(Debug, StructOpt)]
-#[structopt(about = "Tool for displaying and creating C2PA manifests.",global_settings = &[AppSettings::ColoredHelp, AppSettings::ArgRequiredElseHelp])]
+/// Tool for displaying and creating C2PA manifests."
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None, setting = AppSettings::ArgRequiredElseHelp)]
 struct CliArgs {
-    #[structopt(parse(from_os_str))]
-    #[structopt(
-        short = "m",
-        long = "manifest",
-        help = "Path to manifest definition JSON file."
-    )]
+    #[clap(short, long, help = "Path to manifest definition JSON file.")]
```

Comment:
FWIW clap will pick up doc comments (`/// Path to manifest definition JSON file`) which I find more readable.

Reply:
I did it this way here for consistency with all the others, but I can make this one a comment, or change them all to comments.

Reply:
I changed them all to use comments